src/main/java/org/broadinstitute/hellbender/utils/io/TextFeatureReader.java

```diff
@@ -0,0 +1,244 @@
+package org.broadinstitute.hellbender.utils.io;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.tribble.*;
+import htsjdk.tribble.index.Block;
+import htsjdk.tribble.index.Index;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IndexUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.*;
+import java.nio.channels.Channels;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public final class TextFeatureReader<F extends Feature> implements FeatureReader<F> {
+    private final Path path;
+    private final FeatureCodec<F, Iterator<String>> codec;
+    private FeatureCodecHeader header;
+    private Boolean queryable;
+    private Index index;
+
+    private static final long BOGUS_FILE_OFFSET = -1L;
+
+    public TextFeatureReader( final GATKPath path,
+                              final FeatureCodec<F, Iterator<String>> codec ) {
+        this.path = path.toPath();
+        this.codec = codec;
+    }
+
+    @Override
+    public FeatureIterator<F> iterator() {
+        return new FeatureIterator<>(path, codec);
+    }
+
+    @Override
+    public boolean isQueryable() {
+        if ( queryable == null ) {
+            try {
+                queryable = AbstractFeatureReader.isTabix(path.toString(), null);
+            } catch ( final IOException ioe ) {
+                throw new UserException("Error while checking for existence of index for " + path);
+            }
+        }
+        return queryable;
+    }
+
+    @Override
+    public FeatureIterator<F> query( final String chr, final int start, final int end ) {
+        if ( index == null ) {
+            if ( !isQueryable() ) {
+                throw new UserException("Querying non-queryable feature reader for " + path);
+            }
+            index = IndexUtils.loadTabixIndex(path);
+            if ( index == null ) {
+                throw new UserException("Despite its appearing to exist, couldn't load index for " + path);
+            }
+        }
+        final List<Block> blocks = index.getBlocks(chr, start, end);
+        return new FeatureIterator<>(path,
+                codec,
+                new SimpleInterval(chr, start, end),
+                blocks == null || blocks.isEmpty() ? BOGUS_FILE_OFFSET : blocks.get(0).getStartPosition());
+    }
+
+    @Override
+    public void close() {}
+
+    @Override
+    public List<String> getSequenceNames() {
```

Comment:
I'm assuming the codecs this supports don't include a sequence dictionary in their header?  Should they?

Reply:
You noted that the TextFeatureReader returns an empty list in its getSequenceNames method:  Not to despair, this method is not used by anyone.
You asked, "Shouldn't the files that this reader supports include sequence dictionary metadata?".  Answer:  These are legacy SV pipeline file formats.  Might've been nice, but they don't and that horse has left the barn.  Someday we may be able to switch the SV pipeline over to using BlockCompressedInterval files that do have metadata.